js/cards.js

```diff
@@ -107,32 +127,65 @@ function getOwnedCards(cards_data) {
   return ownedCards;
 }
 
-function sortCards(cards, sortType) {
+function sortCards(cards, sortType, reverse=false) {
   sortType = sortType ?? "Collector Number";
-  cards.sort(sort_functions[sortType]);
+  let before = reverse ? -1 : 1
+  let after = reverse ? 1 : -1
+  cards.sort(sort_functions[sortType](before, after));
   return cards;
 }
 
-export function showCollection(cards_data, htmlLocation) {
-  let sort = localStorage.getItem("sort");
+export function showCollection(cards_data, htmlLocation, page = 1) {
+  let sort = localStorage.getItem("sort") ?? "Collector Number";
   let fullCollection = localStorage.getItem("fullCollection") === "true";
+  let cards;
+  let reverse = false;
   let ownedCount = 0;
   for (let rarity of RARITIES) {
     ownedCount += parseInt(localStorage.getItem(`count-${rarity}`) ?? "0");
   }
   COLLECTED_CARDS_NUMBER.textContent = `Collected cards: ${ownedCount}/${cards_data.length}`;
 
+  if (sort[sort.length - 1] === "-") {
+    sort = sort.slice(0, -1)
+    reverse = true;
+  }
+
   if (fullCollection) {
-    let sorted = sortCards(cards_data, sort);
-    renderCards(sorted, htmlLocation, true);
+    cards = sortCards(cards_data, sort, reverse);
   } else {
-    let ownedCards = getOwnedCards(cards_data);
-    if (ownedCards.length == 0) {
+    cards = getOwnedCards(cards_data);
+    if (cards.length == 0) {
       htmlLocation.innerHTML =
         "You have no cards at the moment. Try pulling some at the gacha!";
+      PAGINATION.container.classList.add("hidden");
+      return;
+    } else {
+      cards = sortCards(cards, sort, reverse);
+    }
+  }
+
+  if (cards.length > CARDS_PER_PAGE) {
```

Comment:
(nit) comment explaining this section determines which page buttons to show/ hide?

maybe renaming PAGINATION to PAGINATION_BTNS would help make it clearer too

Reply:
Done o7